gooddata-flight-server/gooddata_flight_server/server/base.py

```diff
@@ -61,3 +62,19 @@ class FlightServerMethodsFactory(Protocol):
     """
 
     def __call__(self, ctx: ServerContext) -> FlightServerMethods: ...
+
+
+class FlightServerMethodsAbstractFactory(abc.ABC):
```

Comment:
the naming here is not correct because this is not actually an abstract factory. eg: https://en.wikipedia.org/wiki/Abstract_factory_pattern

i suggest to go slightly differently in this case; instead of doing 'factory for factory', the server could expose a decorator (used really as a marker) that would be slapped on the existing factory function. the dynamic loading code would search for that decorator.

i don't have experience with this approach but it is definitely possible - other libraries do it too.

simplest alternative is a convention where the factory function must have particular name. the server will look for that function and will fail upon startup otherwise.

Reply:
How is that not an abstract factory? It is a class that returns factories 🤔 
Nevertheless, I like the marker decorator pattern, will give it a go.

Reply:
Rewrote to using decorators.